lib/friendly_id/slugged.rb

```diff
@@ -357,6 +366,10 @@ def set_slug(normalized_slug = nil)
         candidates = FriendlyId::Candidates.new(self, normalized_slug || send(friendly_id_config.base))
         slug = slug_generator.generate(candidates) || resolve_friendly_id_conflict(candidates)
         send "#{friendly_id_config.slug_column}=", slug
+      elsif should_manually_change_friendly_id?
+        candidates = FriendlyId::Candidates.new(self, self.slug)
+        slug = slug_generator.generate(candidates) || resolve_friendly_id_conflict(candidates)
+        send "#{friendly_id_config.slug_column}=", slug
```

Comment:
this could probably be extracted as common functionality:

```ruby
    # Sets the slug.
    def set_slug(normalized_slug = nil)
      if should_generate_new_friendly_id?
        set_slug_with_candidate(normalized_slug || send(friendly_id_config.base))
      elsif should_manually_change_friendly_id?
        set_slug_with_candidate(self.slug)
      end
    end
    private :set_slug

    def set_slug_with_candidate(candidate)
      candidates = FriendlyId::Candidates.new(self, candidate)
      slug = slug_generator.generate(candidates) || resolve_friendly_id_conflict(candidates)
      send "#{friendly_id_config.slug_column}=", slug
    end
    private :set_slug_with_candidate
```

What do you think?